modules/imap/functions.php

```diff
@@ -857,19 +857,31 @@ function imap_refresh_oauth2_token($server, $config) {
  * @return int count of messages moved
  */
 if (!hm_exists('imap_move_same_server')) {
-function imap_move_same_server($ids, $action, $hm_cache, $dest_path) {
+function imap_move_same_server($ids, $action, $hm_cache, $dest_path, $screen_emails=false) {
     $moved = array();
     $keys = array_keys($ids);
     $server_id = array_pop($keys);
     $cache = Hm_IMAP_List::get_cache($hm_cache, $server_id);
     $imap = Hm_IMAP_List::connect($server_id, $cache);
     foreach ($ids[$server_id] as $folder => $msgs) {
         if (imap_authed($imap) && $imap->select_mailbox(hex2bin($folder))) {
-            if ($imap->message_action(mb_strtoupper($action), $msgs, hex2bin($dest_path[2]))) {
+            if ($screen_emails) {
                 foreach ($msgs as $msg) {
                     $moved[]  = sprintf('imap_%s_%s_%s', $server_id, $msg, $folder);
+                    $email = current(array_column(process_address_fld($imap->get_message_headers($msg)['From']), "email"));
+                    $uids = $imap->search('ALL', false, array(array('FROM', $email)));
+                    foreach ($uids as $uid) {
+                        $move = $imap->message_action(mb_strtoupper($action), $uid, hex2bin($dest_path[2]));
```

Comment:
This is done on line 870, if we do it on line 874, we risk having an array that contains more values that are not actually expected at the javascript level. I'm going to say that in the array we put the messages that have been selected by the user, while on line 874 we move the messages that have not been selected.

Reply:
Line 870 adds just the originally selected message to the moved list. You actually move all other messages from the same email address that are found in $uids array. These must go into moved array as well. Otherwise, what happens if user selects 1 message - it says 1 message moved but it actually moved many more from the same sender - it should report the actual number of messages that are moved.

Reply:
Noted

Reply:
Thanks!